org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -135,10 +135,12 @@ public void execute() throws EventException {
 			initializeMalcolmDevice(bean, gen);
 			
 			// We set any activated monitors in the request if none have been specified.
-			if (bean.getScanRequest().getMonitorNames()==null) {
-				Collection<String> dMonNames = getMonitors();
-				bean.getScanRequest().setMonitorNames(dMonNames);
-				logger.debug("Default monitors {}", dMonNames);
+			if (Boolean.getBoolean("org.eclipse.scanning.server.useDefaultActivatedMonitors")) {
+				if (bean.getScanRequest().getMonitorNames()==null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)